whois-api/src/main/java/net/ripe/db/whois/api/fulltextsearch/ElasticFulltextSearch.java

```diff
@@ -151,6 +152,27 @@ private org.elasticsearch.action.search.SearchResponse performFulltextSearch(fin
         }
     }
 
+    private SearchRequest escapeIPv6Colon(final SearchRequest searchRequest){
+        final List<String> ipv6Matches = Ipv6Resource.extractIPv6(searchRequest.getQuery());
+        String originalQuery = searchRequest.getQuery();
+        if (!ipv6Matches.isEmpty()){
+            for (String ipv6Match:ipv6Matches) {
+                originalQuery = originalQuery.replace(ipv6Match, ipv6Match.replace(":", "\\:"));
+            }
+            return new SearchRequest.SearchRequestBuilder()
+                    .setRows(String.valueOf(searchRequest.getRows()))
+                    .setStart(String.valueOf(searchRequest.getStart()))
+                    .setQuery(originalQuery)
```

Comment:
Why are you making a new search request? Can you do it in service class before setting a query

Reply:
I am doing it because there is only a builder that allow change the SearchRequest.
I could see any setters. I don't want to overload the service class I think that this logic should be in a lower layer (service just to throw exception that makes not sense to pass to a lower layer).
I could do that or create a setter and set the value to the current searchRequest

Reply:
Done